cache/mock.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+package cache
+
+import (
+	"github.com/ipfs/go-cid"
+)
+
+// MockCache stores up to size elements, randomly evicting when size is reached
+type MockCache struct {
+	items map[cid.Cid]struct{}
```

Comment:
Any reason for struct instead of something like a bool?

Reply:
`items[cid]` will either be `nil` or `struct{}{}`, as opposed to `nil,` `true` or `false`, takes up a bit less space too I think

Reply:
added a lock and a test

hmm I think `struct{}` works for now but could change it to bool if requested